src/array/fixed_size_list/mutable.rs

```diff
@@ -47,6 +41,11 @@ impl<M: MutableArray> MutableFixedSizeListArray<M> {
             Box::new(Field::new(name, values.data_type().clone(), nullable)),
             size,
         );
+        Self::new_from(values, data_type, size)
+    }
+
+    /// Creates a new [`MutableFixedSizeListArray`] from a [`MutableArray`], [`DataType`] and size.
+    pub fn new_from(values: M, data_type: DataType, size: usize) -> Self {
```

Comment:
I _think_ we need to check that this `data_type` is compatible with `MutableFixedSizeListArray` and `values`, or we end up with an incompatible data_type, right? Something like we do for `ListArray` [here](https://github.com/jorgecarleitao/arrow2/blob/26120da6811eb0b279ebba1dfea2992e66cb1cb2/src/array/list/mod.rs#L59)